src/qt/mAPreferencesWindow.cpp

```diff
@@ -300,21 +386,34 @@ void mAPreferencesWindow::ProbeAudioDevices()
                 ui->audioOutput->setCurrentIndex(ui->audioInput->count()-1);            
         }
     }
-    
+
 //    if( dac == 0 )
 //        ui->audioOutput->setCurrentIndex( 0 );
 //    if( adc == 0 )
 //        ui->audioInput->setCurrentIndex( 0 );
     
-    this->SelectedAudioInputChanged();
-    this->SelectedAudioOutputChanged();
+    this->selectedAudioInputChanged();
+    this->selectedAudioOutputChanged();
+}
+
+
+// 1.5.0.1 (ge) added
+void mAPreferencesWindow::selectedAudioDriverChanged()
+{
+    // get item data
+    int driver = ui->audioDriver->itemData(ui->audioDriver->currentIndex()).toInt();
+    // trigger probe
+    this->probeAudioDevices(driver);
```

Comment:
can force reset to default by adding a bool to `probeAudioDevices`

```
void mAPreferencesWindow::probeAudioDevices( int driver, bool resetToDefault )
{
    // ...
    // load available audio I/O interfaces into the pop up menus
    for(i = 0; i < len; i++)
    {
        if(interfaces[i].outputChannels > 0 || interfaces[i].duplexChannels > 0)
        {
            ui->audioOutput->addItem(interfaces[i].name.c_str(), int(i+1));
            if(i + 1 == dac)
                ui->audioOutput->setCurrentIndex(ui->audioOutput->count()-1);
            if((resetToDefault || dac == 0) && interfaces[i].isDefaultOutput)
                ui->audioOutput->setCurrentIndex(ui->audioOutput->count()-1);
        }

        if(interfaces[i].inputChannels > 0 || interfaces[i].duplexChannels > 0)
        {
            ui->audioInput->addItem(interfaces[i].name.c_str(), int(i+1));
            if(i + 1 == adc)
                ui->audioInput->setCurrentIndex(ui->audioInput->count()-1);
            if((resetToDefault || adc == 0) && interfaces[i].isDefaultInput)
                ui->audioInput->setCurrentIndex(ui->audioInput->count()-1);            
        }
    }
```

Reply:
implemented and corrected the input bugs